sys/sys.go

```diff
@@ -127,3 +129,45 @@ func getSystemBlockDeviceInfo(sysClassBlockDirectory string, readDirFn func(stri
 	}
 	return deviceInfo, nil
 }
+
+// GetBootKernelConfigMap reads the kernel config into a key-value map. It tries to read kernel config from
+// ${bootPrefix}/config-${kernelVersion}, and comments are ignored. If the bootPrefix is empty, it points to /boot by
+// default.
+func GetBootKernelConfigMap(bootPrefix, kernelVersion string) (map[string]string, error) {
+	if kernelVersion == "" {
+		return nil, fmt.Errorf("kernelVersion cannot be empty")
+	}
+	if bootPrefix == "" {
+		bootPrefix = "/boot/"
```

Comment:
Nit: Suggest making this a reusable variable and moving it to the `types` package.